shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,744 @@
+            product = packaging.product_id
+            response = self._use_handlers(
+                handlers, move_line, product, packaging=packaging
+            )
+            if response:
+                return response
+
+    def _set_quantity__valid_dest_location_for_move_line_domain(self, move_line):
+        move_line_dest_location = move_line.location_dest_id
+        return [
+            "|",
+            ("id", "in", move_line_dest_location.ids),
+            ("id", "child_of", move_line_dest_location.ids),
+        ]
+
+    def _set_quantity__valid_dest_location_for_move_line(self, move_line):
+        domain = self._set_quantity__valid_dest_location_for_move_line_domain(move_line)
+        return self.env["stock.location"].search(domain)
+
+    def _valid_dest_location_for_menu_domain(self):
+        return [
+            "|",
+            ("id", "in", self.picking_types.default_location_dest_id.ids),
+            ("id", "child_of", self.picking_types.default_location_dest_id.ids),
+        ]
+
+    def _valid_dest_location_for_menu(self):
+        domain = self._valid_dest_location_for_menu_domain()
+        return self.env["stock.location"].search(domain)
+
+    def _set_quantity__check_location(self, location, move_line, confirmation=False):
+        valid_locations_for_move_line = (
+            self._set_quantity__valid_dest_location_for_move_line(move_line)
+        )
+        valid_locations_for_menu = self._valid_dest_location_for_menu()
+        message = False
+        asking_confirmation = False
+        if location in valid_locations_for_move_line:
+            # scanned location is valid, return no response
+            pass
+        elif (
+            location in valid_locations_for_menu
+            and self.work.menu.allow_alternative_destination
+        ):
+            # Considered valid if scan confirmed
+            if not confirmation:
+                # Ask for confirmation
+                orig_location = move_line.location_dest_id
+                message = self.msg_store.confirm_location_changed(
+                    orig_location, location
+                )
+                asking_confirmation = True
+        else:
+            # Invalid location, return an error
+            message = self.msg_store.dest_location_not_allowed()
+        if message:
+            return self._response_for_set_quantity(
+                move_line, message=message, asking_confirmation=asking_confirmation
+            )
+
+    def _write_destination_on_lines(self, lines, location):
+        # TODO
+        # '_write_destination_on_lines' is implemented in:
+        #
+        #   - 'location_content_transfer'
+        #   - 'zone_picking'
+        #   - 'cluster_picking' (but it is called '_unload_write_destination_on_lines')
+        #
+        # And all of them has a different implementation,
+        # To refactor later.
+        try:
+            # TODO loose dependency on 'shopfloor_checkout_sync' to avoid having
```

Comment:
Why ? there must be a good reason to have a glue module between `shopfloor` and `checkout_sync`.
It does save on code repetition with the locking of related lines https://github.com/OCA/wms/blob/14.0/shopfloor_checkout_sync/actions/checkout_sync.py which is specific.
And separation of concern.

In the implementation of this PR, the locking of lines is missing and probably causes `Serialization Failure`

IMO we should keep here the same form of implementation used in other scenario. And propose a refactor in a distinct PR, if needs be.

Reply:
Ah because this is a specific module and we would need another glue module for checkout sync !
I will fix the missing lock with the current implementation.

Reply:
I just pushed a commit that adds the locking of the lines. And a little test that comes with it